### PR TITLE
doc/baremetal/linker.rst: Add memory map for HSDK and HSDK-4xD

### PR DESCRIPTION
**Pull request by vkremneva**

### CODE REVIEW

File: doc/baremetal/linker.rst [hunk 1]
```diff
@@ -119,6 +119,89 @@ to ``__end_heap``. Therefore heap and stack grow towards each other and eventual
...
+.. code-block:: text
+
+    MEMORY {
+        SYSTEM0  : ORIGIN = 0x00000000, LENGTH = 0x100000000
+        }
+    REGION_ALIAS("startup", SYSTEM0)
+    REGION_ALIAS("text", SYSTEM0)
+    REGION_ALIAS("data", SYSTEM0)
+    REGION_ALIAS("sdata", SYSTEM0)
+    PROVIDE (__stack_top = (0xffffffff & -4 ));
```
Comment (claziss) [inline doc/baremetal/linker.rst]: you don't need this

File: doc/baremetal/linker.rst [hunk 2]
```diff
@@ -119,6 +119,89 @@ to ``__end_heap``. Therefore heap and stack grow towards each other and eventual
...
+    PROVIDE (__end_heap =  (0xffffffff ));
```
Comment (claziss) [inline doc/baremetal/linker.rst]: nor this

File: doc/baremetal/linker.rst [hunk 3]
```diff
@@ -119,6 +119,89 @@ to ``__end_heap``. Therefore heap and stack grow towards each other and eventual
...
+        CCMWRAP0 : ORIGIN = 0x70040000, LENGTH = 0x0ffc0000
+        DCCM     : ORIGIN = 0x80000000, LENGTH = 0x00040000
+        CCMWRAP1 : ORIGIN = 0x80040000, LENGTH = 0x0ffc0000
+        SYSTEM1  : ORIGIN = 0x90000000, LENGTH = 0x70000000
+        }
+    REGION_ALIAS("startup", ICCM0)
+    REGION_ALIAS("text", ICCM0)
+    REGION_ALIAS("data", DCCM)
+    REGION_ALIAS("sdata", DCCM)
+    PROVIDE (__stack_top = (0x8003ffff & -4 ));
```
Comment (claziss) [inline doc/baremetal/linker.rst]: same

File: doc/baremetal/linker.rst [hunk 4]
```diff
@@ -119,6 +119,89 @@ to ``__end_heap``. Therefore heap and stack grow towards each other and eventual
...
+        SYSTEM0  : ORIGIN = 0x00000000, LENGTH = 0xb0000000
+        CSM      : ORIGIN = 0xb0000000, LENGTH = 0x00040000
+        CCMWRAP0 : ORIGIN = 0xb0040000, LENGTH = 0x0ffc0000
+        SYSTEM1  : ORIGIN = 0xc0000000, LENGTH = 0x40000000
+        }
+    REGION_ALIAS("startup", SYSTEM0)
+    REGION_ALIAS("text", SYSTEM0)
+    REGION_ALIAS("data", SYSTEM0)
+    REGION_ALIAS("sdata", SYSTEM0)
+    PROVIDE (__stack_top = (0xafffffff & -4 ));
```
Comment (claziss) [inline doc/baremetal/linker.rst]: same

File: doc/baremetal/linker.rst [hunk 5]
```diff
@@ -119,6 +119,89 @@ to ``__end_heap``. Therefore heap and stack grow towards each other and eventual
...
+        SYSTEM4  : ORIGIN = 0x70000000, LENGTH = 0x40000000
+        CSM      : ORIGIN = 0xb0000000, LENGTH = 0x00040000
+        CCMWRAP2 : ORIGIN = 0xb0040000, LENGTH = 0x0ffc0000
+        SYSTEM5  : ORIGIN = 0xc0000000, LENGTH = 0x40000000
+        }
+    REGION_ALIAS("startup", ICCM0)
+    REGION_ALIAS("text", ICCM0)
+    REGION_ALIAS("data", DCCM)
+    REGION_ALIAS("sdata", DCCM)
+    PROVIDE (__stack_top = (0x6000ffff & -4 ));
```
Comment (claziss) [inline doc/baremetal/linker.rst]: same

**Review by claziss — CHANGES_REQUESTED** [review]

File: doc/baremetal/linker.rst (hunk 5 above)
Comment (vkremneva) [reply doc/baremetal/linker.rst]: Should I also remove those sections from previous version of this file? https://github.com/foss-for-synopsys-dwc-arc-processors/toolchain/blame/arc-releases/doc/baremetal/linker.rst#L88

**Comment by vkremneva** [comment]
@claziss I made changes that you requested, could you please take a look?

**Review by claziss — APPROVED** [review]

**Review by abrodkin — APPROVED** [review]

**Pull request merged**